### DISCUSSION TITLE
`default-x` for `<words>`

### DISCUSSION THREAD

**Discussion by lemzwerg** [w3c-cg/musicxml#558] [Q&A] [answered]
Consider this small MusicXML snippet.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 4.0 Partwise//EN"
                                "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise version="4.0">
  <part-list>
    <score-part id="P1"><part-name></part-name></score-part>
  </part-list>
  <part id="P1">
    <measure number="1">
      <attributes><divisions>1</divisions></attributes>
      <direction placement="above">
        <direction-type>
          <words default-x="50" default-y="50">(50,50)</words>
        </direction-type>
      </direction>
      <direction placement="below">
        <direction-type>
          <words default-x="-50" default-y="-150">(-50,-150)</words>
        </direction-type>
      </direction>
      <note>
        <pitch><step>F</step><octave>4</octave></pitch>
        <duration>4</duration>
        <type>whole</type>
      </note>
    </measure>
  </part>
</score-partwise>
```

Here are the rendering results of Finale and MuseScore:

![image](https://github.com/user-attachments/assets/28bb4820-41e4-44a1-a32c-770cdb5c692d)

![image](https://github.com/user-attachments/assets/81d0f72d-5b1a-49c4-a570-c928c718d180)

I've also tested Verovio and SoundSlice.  Only Finale honours `default-y`, while all four programs ignore `default-x`.  Apparently, this sentence from MusicXML's description of `default-x`,

> This attribute provides higher-resolution positioning data than the `<offset>` element. Applications reading a MusicXML file that can understand both features should generally rely on this attribute for its greater accuracy.

is not popular while importing MusicXML...  Interestingly, the files in the `musicxml-samples` bundle are full with `default-x` attributes for `<words>`.

My conclusion is that `default-x` and `default-y` are completely unreliable for MusicXML interchange.  While this is sort-of acceptable for the vertical axis, it is really problematic IMHO for horizontal positioning of words – users tend to move word items wildly around, probably not taking care of correctly anchoring them to the right notes.  Am I correct or am I doing something wrong?  Shall I only rely on `<offset>`?

**Comment by Jojo-Schmitz** [comment]
Just for the record: this is how MuseScore 2.3.2 renders it:
![image](https://github.com/user-attachments/assets/6d310bf1-0a43-4707-8276-f2d948a738e3)
And this how MuseScore 1.3 does it:
![image](https://github.com/user-attachments/assets/c94a0eb4-ef12-4e86-aa36-4600e2409377)
You're right though for 3.6.2 and later, up to 4.5dev

**Reply by lemzwerg** [reply]
Thanks for the detailed information!

**Comment by mdgood** [comment, marked as answer]
The `<offset>` element is indeed generally better for horizontal positioning. It better matches the way many, if not most, notation applications store this information.

**Reply by lemzwerg** [reply]
Thanks for confirmation.
